File: CPP/Program/Mouse.cpp
#ifndef MOUSE_CPP
#define MOUSE_CPP

#include <iostream>
#include <string>
#include "Periferal.cpp"
using namespace std;

class Mouse : public Periferal {
private:
    int dpi;
    int jumlahButton;
    
public:
    Mouse() {
    }
    
    Mouse(int dpi, int jumlahButton, string jenis, string koneksi) : Periferal(jenis, koneksi) {
        this->dpi = dpi;
        this->jumlahButton = jumlahButton;
    }
    
    void setDpi(int dpi) {
        this->dpi = dpi;
    }
    
    void setJumlahButton(int jumlahButton) {
        this->jumlahButton = jumlahButton;
    }
    
    int getDpi() {
        return this->dpi;
    }
    
    int getJumlahButton() {
        return this->jumlahButton;
    }
    
    ~Mouse() {
    }
};

#endif

File: Python/Program/Mouse.py
from Periferal import Periferal

class Mouse(Periferal):
    def __init__(self, dpi=0, jumlah_button=0, jenis="", koneksi=""):
        super().__init__(jenis, koneksi)
        self.__dpi = dpi
        self.__jumlah_button = jumlah_button
    
    def set_dpi(self, dpi):
        self.__dpi = dpi
    
    def set_jumlah_button(self, jumlah_button):
        self.__jumlah_button = jumlah_button
    
    def get_dpi(self):
        return self.__dpi
    
    def get_jumlah_button(self):
        return self.__jumlah_button